test/e2e/app/app.go

```diff
@@ -852,12 +847,7 @@ func (app *Application) verifyAndSum(
 		if err != nil {
 			return 0, fmt.Errorf("could not hex-decode public key for validator address %s, err %w", valAddr, err)
 		}
-		var pubKeyProto cryptoproto.PublicKey
-		err = pubKeyProto.Unmarshal(pubKeyBytes)
-		if err != nil {
-			return 0, fmt.Errorf("unable to unmarshal public key for validator address %s, err %w", valAddr, err)
-		}
-		pubKey, err := cryptoenc.PubKeyFromProto(pubKeyProto)
+		pubKey, err := cryptoenc.PubKeyFromTypeAndBytes(app.cfg.KeyType, pubKeyBytes)
```

Comment:
This comment is symmetric to the one above. You shouldn't use `app.cfg.KeyType` here, but rather get it from the deserialized data.
If we use `app.cfg.KeyType` here, we won't be able to test a validator set with several keytypes